File: src/main/resources/static/js/model/Payment.js
namespace("bg.infa.pbt.model");

bg.infa.pbt.model.Payment = function(dto) {
	this.id = dto.id !== undefined ? dto.id : null;
	this.budgetCategory = dto.budgetCategory !== undefined ? new bg.infa.pbt.model.BudgetCategory(dto.budgetCategory) : null;
	this.monthNumber = dto.monthNumber !== undefined ? dto.monthNumber : null;
	this.amount = dto.amount !== undefined ? dto.amount : null;
};

File: src/main/java/bg/infa/pbt/budget/Payment.java
package bg.infa.pbt.budget;

public class Payment {
	private static int seqId;

	private int id;
	private BudgetCategory budgetCategory;
	private int amount;
	private int monthNumber;

	public Payment(BudgetCategory budgetCategory, int amount, int monthNumber) {
		this.id = seqId++;
		this.budgetCategory = budgetCategory;
		this.amount = amount;
		this.monthNumber = monthNumber;
	}

	public int getId() {
		return id;
	}

	public BudgetCategory getBudgetCategory() {
		return budgetCategory;
	}

	public void setBudgetCategory(BudgetCategory budgetCategory) {
		this.budgetCategory = budgetCategory;
	}

	public int getAmount() {
		return amount;
	}

	public void setAmount(int amount) {
		this.amount = amount;
	}

	public int getMonthNumber() {
		return monthNumber;
	}

	public void setMonthNumber(int monthNumber) {
		this.monthNumber = monthNumber;
	}
}
